libraries/mbed/targets/hal/TARGET_STM/TARGET_STM32F3/pwmout_api.c

```diff
@@ -161,8 +161,24 @@ void pwmout_period_us(pwmout_t* obj, int us)
     // Update the SystemCoreClock variable
     SystemCoreClockUpdate();
 
-    TimHandle.Init.Period        = us - 1;
-    TimHandle.Init.Prescaler     = (uint16_t)(SystemCoreClock / 1000000) - 1; // 1 us tick
+    /* To make it simple, we use to possible prescaler values which lead to:
+     * pwm unit = 1us, period/pulse can be from 1us to 65535us
+     * or
+     * pwm unit = 500us, period/pulse can be from 500us to ~32.76sec
+     * Be careful that all the channels of a PWM shares the same prescaler
+     */
+    if (us >  0xFFFF) {
+        obj->prescaler = 500;
+    } else {
+        obj->prescaler = 1;
+    }
+    TimHandle.Init.Prescaler     = ((SystemCoreClock / 1000000) * obj->prescaler) - 1;
+
+    MBED_ASSERT(TimHandle.Init.Prescaler < 0xFFFF);
```

Comment:
I would suggest using ``error()``, same for the one below, as this is intended to be run-time error?